org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceDiagnosticsHandler.java

```diff
@@ -137,7 +141,16 @@ private static Diagnostic toDiagnostic(IDocument document, IMarker marker) {
 		d.setSource(JavaLanguageServerPlugin.SERVER_SOURCE_ID);
 		d.setMessage(marker.getAttribute(IMarker.MESSAGE, ""));
 		d.setCode(String.valueOf(marker.getAttribute(IJavaModelMarker.ID, 0)));
-		d.setSeverity(convertSeverity(marker.getAttribute(IMarker.SEVERITY, -1)));
+		try {
+			if (IJavaModelMarker.TASK_MARKER.equals(marker.getType())) {
+				d.setSeverity(DiagnosticSeverity.Warning);
```

Comment:
it's probably controversial but I think TODOs should have severity INFO (which we might make configurable eventually).
Now if the only easy way to have consistent severity is to set it to WARNING, I can live with that for starters. 
We can probably add preferences later.